keps/sig-api-machinery/20200123-client-go-ctx.md

```diff
@@ -94,6 +92,48 @@ func (c *pods) Get(ctx context.Context, name string, options meta_v1.GetOptions)
 Here we extend the signature to add `context.Context` as the first argument,
 then attach the context to the request.
 
+### Cleanup Inconsistent \*Options Passing
+
+While we are requiring a migration for consumers of the client, now is a good
+time to fix any issues we have with the API we are deprecating. While lack of
+context support is the primary motiviation, clientset methods do not
+consistently support passing \*Options in the current API. Take for example the
+signatures of the existing flunders client:
+
+```
+func (c *flunders) Get(string, metav1.GetOptions) (*v1beta1.Flunder, error)
+func (c *flunders) List(metav1.ListOptions) (*v1beta1.FlunderList, error)
+func (c *flunders) Watch(metav1.ListOptions) (watch.Interface, error)
+func (c *flunders) Create(*v1beta1.Flunder) (*v1beta1.Flunder, error)
+func (c *flunders) Update(*v1beta1.Flunder) (*v1beta1.Flunder, error)
+func (c *flunders) UpdateStatus(*v1beta1.Flunder) (*v1beta1.Flunder, error)
+func (c *flunders) Delete(string, *metav1.DeleteOptions) error
+func (c *flunders) DeleteCollection(*metav1.DeleteOptions, metav1.ListOptions) error
+func (c *flunders) Patch(name string, pt types.PatchType, data []byte, subresources ...string) (*v1beta1.Flunder, error)
+```
+
+1. Create does not accept a metav1.CreateOptions.
+1. Update, UpdateStatus and Patch do not accept metav1.UpdateOptions.
+1. Delete methods accept metav1.DeleteOptions by pointer reference.
+
+TODO(mikedanese): Which of the inconsistencies do we want to fix in this round?
```

Comment:
@liggitt @deads2k, which of these issues do we want to fix here? Are there any more?

Reply:
All `*Options` should be specifiable. I haven't swept to see if there is any behavior difference between nil and `&metav1.DeleteOptions{}` from a client's perspective, but I would lean toward non-pointer structs if the behavior is equivalent